cpp/modules/deck.gl/core/src/arrow/arrow-mapper.cc

```diff
@@ -24,6 +24,29 @@
 
 using namespace deckgl;
 
+auto ArrowMapper::mapBoolColumn(const std::shared_ptr<arrow::Table>& table, std::function<BoolAccessor> getValueFromRow)
+    -> std::shared_ptr<arrow::Array> {
+  arrow::MemoryPool* pool = arrow::default_memory_pool();
+  arrow::BooleanBuilder builder{pool};
+
+  for (auto i = 0; i < table->num_rows(); ++i) {
+    // TODO(ilija@unfolded.ai): Move row object creation out of the loop
+    auto row = Row{table, i};
+    auto value = getValueFromRow(row);
+
+    if (!builder.Append(value).ok()) {
```

Comment:
Call `Reserve` on the builder before appending values? I assume otherwise this works like a regular array builder and needs to resize it's buffer.

Reply:
Correct, wasn't sure what a reasonable chunk to reserve would be, so I postponed it until indexing was in place as the number of rows has decreased.

Added a conservative estimate before we start iterating